src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved. 
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Benchmarks.Configuration;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Benchmarks.Middleware
+{
+    public class MemoryCachePlaintextMiddleware
+    {
+        private static readonly PathString _path = new PathString(Scenarios.GetPath(s => s.MemoryCachePlaintext));
+        private static readonly object _key = new object();
+
+        private readonly RequestDelegate _next;
+        private readonly IMemoryCache _memoryCache;
+
+        public MemoryCachePlaintextMiddleware(RequestDelegate next, IMemoryCache memoryCache)
+        {
+            _next = next;
+            _memoryCache = memoryCache;
+        }
+
+        public Task Invoke(HttpContext httpContext)
+        {
+            if (httpContext.Request.Path.StartsWithSegments(_path, StringComparison.Ordinal))
+            {
+                byte[] payload;
+                if (!_memoryCache.TryGetValue(_key, out payload))
+                {
+                    payload = Encoding.UTF8.GetBytes("Hello, World!");
+
+                    var cacheEntry = _memoryCache.CreateEntry(_key);
+                    cacheEntry.Value = payload;
```

Comment:
Yes, i will add a 1-second expiration for parity with the ResponseCaching scenario.